src/renderer/MenuRenderer.h

```diff
@@ -0,0 +1,152 @@
+#ifndef MENU_RENDERER_H
+#define MENU_RENDERER_H
+
+#include "display/DisplayInterface.h"
+#include <Arduino.h>
+#include <utils/utils.h>
+
+/**
+ * @class MenuRenderer
+ * @brief Abstract base class for rendering a menu on a display.
+ *
+ * This class provides the basic structure and functionality for rendering a menu
+ * on a display. It includes methods for drawing items, moving the cursor, and
+ * handling edit mode.
+ */
+class MenuRenderer {
+    friend class MenuScreen;
+
+  protected:
+    const uint8_t maxCols;
+    const uint8_t maxRows;
+
+    uint8_t cursorCol;
+    uint8_t cursorRow;
+
+    uint8_t activeRow;
+
+    uint8_t blinkerPosition;
+
+    uint8_t itemIndex;
+    uint8_t itemCount;
+
+    bool inEditMode;
+
+    unsigned long startTime = 0;
+
+  public:
+    /**
+     * Pointer to the display interface used for rendering.
+     */
+    DisplayInterface* display;
+
+    /**
+     * @brief Constructor for MenuRenderer.
+     * @param display Pointer to the display interface.
+     * @param maxCols Maximum number of columns in the display.
+     * @param maxRows Maximum number of rows in the display.
+     */
+    MenuRenderer(DisplayInterface* display, uint8_t maxCols, uint8_t maxRows);
+
+    /**
+     * @brief Initializes the display and starts the timer.
+     */
+    virtual void begin();
+
+    /**
+     * @brief Function to draw a byte on the display.
+     * @param byte The byte to be drawn.
+     */
+    virtual void draw(uint8_t byte) = 0;
+
+    /**
+     * @brief Draws an item on the display.
+     * @param itemIndex Index of the item to be drawn.
+     * @param screenRow Row on the screen where the item should be drawn.
+     * @param text Text of the item to be drawn.
+     */
+    virtual void drawItem(uint8_t itemIndex, uint8_t screenRow, const char* text);
```

Comment:
Can `itemIndex` be avoided. Looks like leak of internal information of screen.
Maybe add methods like `markUpScroll`, `markDownScroll` to decouple item rendering from arrows rendering?

Reply:
#234